doct: Add how to for workers/schedulers (#2448)
canonical/charmcraft: docs/howto/manage-web-app-charms/use-web-app-charm.rst

To obtain a TLS certificate signed by a trusted certificate authority (CA)
using the ACME protocol, use the `LEGO <https://charmhub.io/lego>`_
charm instead.

Run a worker or scheduler process in your workloads
---------------------------------------------------

You can run extra processes in the workload charm that receive the same environment

> variables as the main application.

Author: mmm not sure, it is the last sentence in the section. Do you think I could put it as the first one instead?

> variables as the main application.

Reviewer: Yes, I think moving the last sentence of the section to here makes sense. But I also think it's worth emphasizing in this paragraph what file the user needs to update.

A common use case is running `Celery beat`_
as a scheduler to kick off tasks at regular intervals, together with
`Celery worker servers`_ to execute those tasks.
Celery beat should run in only one unit, while multiple Celery worker
servers can be used.

> For a Flask application that has the Celery app in the module
> ``webapp.app.celery_app``, you can add a snippet like the following:
> 
> .. code-block:: yaml
> 

Reviewer: ```suggestion
For a Flask application that has the Celery app in the module
``webapp.app.celery_app``, you can add a snippet like the following
into ``charmcraft.yaml``:

.. code-block:: yaml
    :caption: charmcraft.yaml

```
Just trying to make it clear what file the user is updating.

    services:
      celery-worker:
        override: replace
        command: celery -A webapp.app.celery_app worker --loglevel=INFO
        startup: enabled
        user: _daemon_
        working-dir: /flask/app
      celery-beat-scheduler:
        override: replace
        command: celery -A webapp.app.celery_app beat -s /tmp/celerybeat-schedule
        startup: enabled
        user: _daemon_
        working-dir: /flask/app

> The service name suffixed with ``-worker`` will run in all units of the application,

Reviewer: ```suggestion
The service name suffixed with ``-worker`` will run in all units of the app,
```

To obtain a TLS certificate signed by a trusted certificate authority (CA)
using the ACME protocol, use the `LEGO <https://charmhub.io/lego>`_
charm instead.

Run a worker or scheduler process in your workloads
---------------------------------------------------

> You can run extra processes in the workload charm that receive the same environment
> variables as the main app. These services must be added to the
> ```rockcraft.yaml`` file under the ``services`` section. If the service name
> ends with ``-worker`` it will run in all units of the app,
> while if the name ends with ``-scheduler`` it will run in only one unit.

Reviewer 2: ```suggestion
You can run extra processes in the workload charm that receive the same environment
variables as the main app. These services must be added to the
project file as entries of the ``services`` key. If the service name
ends with ``-worker`` it will run in all units of the app,
while if the name ends with ``-scheduler`` it will run in only one unit.
```

A common use case is running `Celery beat`_
as a scheduler to kick off tasks at regular intervals, together with
`Celery worker servers`_ to execute those tasks.
Celery beat should run in only one unit, while Celery worker servers
will run in all units.

For a Flask app that has the Celery app in the module

> ``webapp.app.celery_app``, you can add a snippet like the following
> into ``rockcraft.yaml``:

Reviewer 2: ```suggestion
``webapp.app.celery_app``, you can declare them as services like this:
```
